utils/lib/DataSource.ts

```diff
@@ -23,13 +24,42 @@ export interface DataSourceMutationResponse {
   };
 }
 
+interface DataSourceContext extends QuickstartContext {}
+
 class DataSource extends Component<DataSourceConfig, string> {
+  private isCoreDataSource: boolean = false;
+
+  constructor(
+    identifier: string,
+    basePath: string = path.join(__dirname, '..', '..'),
+    context: DataSourceContext = {}
+  ) {
+    super(identifier, basePath);
+    const coreDataSourceIds = context?.coreDataSourceIds ?? []
+    this.isCoreDataSource = coreDataSourceIds?.includes(identifier) ?? false;
+
+    // The `super()` constructor gets called prior to `this.isCoreDataSource`
+    // is defined. Due to this, the instance of a data source will
+    // initially be marked as `invalid`.
+    if (this.isCoreDataSource) {
+      this.identifier = identifier;
+      this.isValid = true;
+
+      // Now that isCoreDataSource is defined, we want to construct our config
+      this.config = this.getConfigContent();
+    }
+  }
+
   /**
    * @returns Filepath for the configuration file (from top-level directory).
    */
   getConfigFilePath() {
     const id = this.identifier;
 
+    if (this.isCoreDataSource) {
+      return id;
+    }
```

Comment:
This may be something we do elsewhere, but I think this makes things a little confusing. This function is `getConfigFilePath` but if that data source is CORE, then it returns an ID. We're kinda abusing the fact that this function returns a string. I could see it being far more clear if it returned the empty string in this case. Let me know what you think, just spit balling here.

Reply:
Yeah, thats a great callout. 

I kept that portion in before "revalidating" the data source in the constructor (lines 39-52). Sincethe quickstart class only checks to see if the data source `isValid`.

`getConfigFilePath` doesn't actually get re-ran, so this block of code _never_ gets executed. 

Removed!